File: src/main/scala/com/github/nichijou/utils/RandomArr.scala
package com.github.nichijou.utils

import scala.util.Random

/**
  * Created by zk_chs on 16/4/26.
  */
object RandomArr {

  def shuffle(length: Int) = {
    if (length <= 0) Array
    val random = Random
    val arr = new Array[Int](length)
    arr(0) = 0
    for (i <- 1 until length) {
      val randomIndex = random.nextInt(i + 1)
      arr(i) = arr(randomIndex)
      arr(randomIndex) = i
    }
    arr
  }

  def main(args: Array[String]) {
    val arr = shuffle(10)
    for (elem <- arr) {print(elem)}
  }

}

File: src/main/java/com/github/algorithms/RandomArr.java
package com.github.algorithms;

import java.util.Arrays;
import java.util.Random;

/**
 * Created by zk_chs on 16/4/28.
 */
public class RandomArr {

    public static Integer[] shuffle (int length){
        if (length <= 0) return new Integer[0];
        Integer[] arr = new Integer[length];
        Random random = new Random();
        arr[0] = 0;
        for (int i = 1; i < length; i++){
            int randomIndex = random.nextInt(i + 1);
            arr[i] = arr[randomIndex];
            arr[randomIndex] = i;
        }
        return arr;
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(shuffle(30)));
    }

}
